### ISSUE TITLE
Problems Runing Cups on Synology as a docker

### ISSUE THREAD

**Issue by hemant5400z** [olbat/dockerfiles#48]
Hi Olbat (and all),

I'm trying to run Cups in a docker and the launch goed fine, setp also works and i can see the printer on my Iphone.
When I launch a print job I can select the printer (using Gutenberg driver HP3800dn).

The problems comes when actually printing, it somtimes start printing but stops after one page or does not rpint at all.

Maybe you can share your expertise on htis issue?

Hemant

**Comment by hemant5400z** [comment]
I Also tried the Postscript driver and the first job goes ok.
But on the next job I get "Unable to write print data: Broken pipe"

after second page.

**Comment by hemant5400z** [comment]
lpstat shows device
and scheduler is also started

**Comment by olbat** [comment]
Hi @hemant5400z,

> Maybe you can share your expertise on htis issue?

I'm going to be honest with you, I don't have much 😄 

It looks like a CUPS-related issue so I'd suggest you to reach out to [the team](https://tracker.debian.org/pkg/cups) that maintains the Debian package, [the maintainers](https://github.com/OpenPrinting/cups/#getting-support-and-other-resources) of the upstream project or give a shot to a more generic support channel (Linux IRC chatroom, Debian forum, ...) to troubleshoot.

Good luck with your troubleshooting & don't hesitate to share any findings!

**Comment by hemant5400z** [comment]
Hi,

I thought it was indeed but i installed the same version manually (no docker) and i don’t have any issues and can print without any problems.

The problem occurs when using synology and docker.

I tried lpd and socket both give the same problem.

So looks like something is missing when passing print jobs through synology and docker.
Hemant

Verstuurd vanaf mijn iPhone

Op 10 feb. 2021 om 17:56 heeft Luc Sarzyniec <notifications@github.com> het volgende geschreven:

﻿

Hi @hemant5400z<https://github.com/hemant5400z>,

Maybe you can share your expertise on htis issue?

I'm going to be honest with you, I don't have much 😄

It looks like a CUPS-related issue so I'd suggest you to reach out to the team<https://tracker.debian.org/pkg/cups> that maintains the Debian package, the maintainers<https://github.com/OpenPrinting/cups/#getting-support-and-other-resources> of the upstream project or give a shot to a more generic support channel (Linux IRC chatroom, Debian forum, ...) to troubleshoot.

Good luck with your troubleshooting & don't hesitate to share any findings!

**Comment by olbat** [comment]
Are you sure that the version & the runtime environment are exactly the same? (same version of Debian, same configuration, same packages installed, etc.).

**Comment by olbat** [comment]
Could the broken pipe issue be related to some firewall / Synology configuration issue?

**Comment by hemant5400z** [comment]
Firewall not sure why wouldnit print 2 pages.

I allowed 5353 and 631 to tje firewall for mu subnet and to be sure also for the bridged 172.zxx.

But it gave the same error.

You are right runtime is not 100 % as mij virtual machine is based on ubuntu.

Only cups version is the same.

Verstuurd vanaf mijn iPhone

Op 10 feb. 2021 om 19:18 heeft Luc Sarzyniec <notifications@github.com> het volgende geschreven:

﻿

Could the broken pipe issue be related to some firewall / Synology configuration issue?

**Comment by olbat** [comment]
Could you try to install the same version of the drivers than the ones that managed to get working?

**Comment by hemant5400z** [comment]
For cups you mean?

It is version 2.3.3 with gutenprint 5.3 which i recall is the same as jn the docker.

Verstuurd vanaf mijn iPhone

Op 10 feb. 2021 om 20:44 heeft Luc Sarzyniec <notifications@github.com> het volgende geschreven:

﻿

Could you try to install the same version of the drivers than the ones that managed to get working?

**Comment by olbat** [comment]
> For cups you mean?

Not really, I'm speaking about the drivers for your printer (that are used by CUPS).

**Comment by hemant5400z** [comment]
Hi,

I used the exact same ones as in the docker; Gutenprint 5.3.3 HP3800DN

Verzonden vanuit Mail<https://go.microsoft.com/fwlink/?LinkId=550986> voor Windows 10

Van: Luc Sarzyniec<mailto:notifications@github.com>
Verzonden: donderdag 11 februari 2021 18:16
Aan: olbat/dockerfiles<mailto:dockerfiles@noreply.github.com>
CC: hemant5400z<mailto:hemant_5400z@hotmail.com>; Mention<mailto:mention@noreply.github.com>
Onderwerp: Re: [olbat/dockerfiles] Problems Runing Cups on Synology as a docker (#48)


For cups you mean?

Not really, I'm speaking about the drivers for your printer (that are used by CUPS).

**Comment by hemant5400z** [comment]
Hi Luc,

I have been troubleshooting for a while and seems alls is working now on a Synology running DSM 6.2.3.
After I found that  for the HP3800DN  the Foomatic drivers working i ran into another issue; while the printer was issible on IOS and the test page from cups admin was sucessfull i always got the message " printer offline".

So I started the docker slightly different 

expsing both 631 and 5353 and not using --host (trial on error). For some reason now it works both PCL and Postcript in color and duplex.

Now i can use the synology as a print server!

cheers,
Hemant
